### ISSUE TITLE
validate on blur for class component

### ISSUE THREAD

**Issue by chipit24** [christianalfoni/formsy-react#392]
I'm writing my React components using the new ES6 class syntax and the `Formsy.HOC` method. I've tried implementing [validate on blur](https://github.com/christianalfoni/formsy-react/wiki/validate-on-blur), but have been unable to get it to work.

Whenever I try typing into the input, no values get set.
Here is my code:

```js
import React, { Component } from 'react';
import { HOC } from 'formsy-react';

class FormInput extends Component {
  changeValue(event) {
    if (this.props.getErrorMessage() != null) {
      this.props.setValue(event.currentTarget.value);
    } else {
      if (this.props.isValidValue(event.target.value)) {
        this.props.setValue(event.target.value);
      } else {
        this.setState({
          _value: event.currentTarget.value,
          _isPristine: false
        });
      }
    }
  }
  
  blurValue(event) {
    this.props.setValue(event.currentTarget.value);
  }
  
  render() {
    return (
      <div>
        <input type="text" onBlur={this.blurValue.bind(this)} onChange={this.changeValue.bind(this)} value={this.props.getValue() || ''} />
    );
  }
};

export default HOC(FormInput);
```

I'm sure the example on the wiki works using mixins, but how can I get it to work with the setup I have above?

**Comment by mattfordham** [comment]
I too am having this issue. It seems setting _value within state should have some effect, but it doesn't when using HOC.

**Comment by chipit24** [comment]
`this` inside `Mixin.js` refers to a different scope than `this` when called within the HOC wrapped component.

**Comment by GautierT** [comment]
+1. HOC doesn't behave like mixin and i don't know why. So i have to switch to classic `React.createClass` for now.

**Comment by Extra-lightwill** [comment]
+1

**Comment by aesopwolf** [comment]
Give this a try:

```jsx
import React, { Component } from 'react';
import { HOC } from 'formsy-react';

class FormInput extends Component {
  constructor(props) {
    super(props);

    this.state = {
      blurred: false,
    };
  }

  handleBlur() {
    this.setState({
      blurred: true,
    })
  }

  handleFocus() {
    this.setState({
      blurred: false,
    })
  }

  render() {
    var errorMessageFromBluring = this.state.blurred ? this.props.getErrorMessage() : '';
    return (
      <div>
        <p>{errorMessageFromBluring}</p>
        <input
          type="text"
          onFocus={this.handleFocus.bind(this)}
          onBlur={this.handleBlur.bind(this)}
          onChange={e => this.props.setValue(e.target.value)}
          value={this.props.getValue() || ''}
        />
      </div>
    );
  }
}

export default HOC(FormInput);
```

**Comment by Extra-lightwill** [comment]
@aesopwolf Yep, cool this works for the error message - but I also want a red outline around my form input when there is invalid input - in the render method in the code below, there is a color conditional (the line directly below 'let errorMessageFromBlurring') - this code works in my es5 version of the form but not here (the input gets a red outline as soon as i start typing) and I'm not sure why? Any ideas how to fix this using this method?

`import React, { Component } from 'react';
import { HOC } from 'formsy-react';

class DefaultInputHOC extends Component {
  constructor(props) {
    super(props);

    this.handleBlur = this.handleBlur.bind(this);
    this.handleFocus = this.handleFocus.bind(this);
    this.keyDown = this.keyDown.bind(this);

    this.state = {
      blurred: false,
    };
  }

  handleBlur() {
    this.setState({
      blurred: true,
    })
  }

  handleFocus() {
    this.setState({
      blurred: false,
    })
  }

  keyDown (e){
        if(e.keyCode=='13'){
            this.setValue(e.currentTarget.value);
        }
    }

  render() {
    let errorMessageFromBlurring = this.state.blurred ? this.props.getErrorMessage() : '';

         let color = '#D9D9D9';
    if (this.props.getErrorMessage()) {
      color = 'red';
    }

    let inputStyleDefault = {
      border: '1px solid ' + color,
      borderRadius: 4,
      outline: 'none',
      paddingLeft: 10,
      width: 1000,
      height: 40,
      marginTop: 28,
    };

    return (
      <div>
        <input
          style={inputStyleDefault} 
          type="text"
          onFocus={this.handleFocus}
          onBlur={this.handleBlur}
          onChange={e => this.props.setValue(e.target.value)}
          value={this.props.getValue() || ''}
          keyDown={this.keyDown}
        />
        <p style={styles.errorMessage}>{errorMessageFromBlurring}</p>
      </div>
    );
  }
}

const styles = {
    errorMessage: {
    color: 'red',
    marginTop: 6,
  },
}

export default HOC(DefaultInputHOC);`
